OpenSim/Simulation/Test/testModelInterface.cpp

```diff
@@ -0,0 +1,133 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#include <OpenSim/Auxiliary/auxiliaryTestFunctions.h>
+#include <OpenSim/Simulation/Model/Model.h>
+#include <OpenSim/Simulation/Model/PhysicalOffsetFrame.h>
+#include <OpenSim/Simulation/Manager/Manager.h>
+#include <OpenSim/Common/LoadOpenSimLibrary.h>
+
+using namespace OpenSim;
+using namespace std;
+
+
+int main() {
+    LoadOpenSimLibrary("osimActuators");
+
+    try {
+        // the specific model used here is not important
+        Model model("arm26.osim");
+        model.finalizeFromProperties();
+
+        // model should be up-to-date with its properties
+        ASSERT(model.isObjectUpToDateWithProperties());
+
+        // make an edit to a Component contained in the model's Set
+        model.updMuscles()[0].upd_min_control() = 0.02;
+        ASSERT(!model.isObjectUpToDateWithProperties());
+
+        model.finalizeFromProperties();
+        ASSERT(model.isObjectUpToDateWithProperties());
+
+        // make an edit to a Component contained in another Set
+        model.updBodySet()[1].upd_mass() = 0.05;
+        ASSERT(!model.isObjectUpToDateWithProperties());
+
+        model.finalizeFromProperties();
+        ASSERT(model.isObjectUpToDateWithProperties());
+
+        // make an edit through model's ComponentList access
+        for (auto& body : model.updComponentList<Body>()) {
+            body.upd_mass_center() = SimTK::Vec3(0);
+        }
+        ASSERT(!model.isObjectUpToDateWithProperties());
+
+        SimTK::State dummy_state;
+
+        ASSERT_THROW(ComponentHasNoSystem, model.getSystem());
+        ASSERT_THROW(ComponentHasNoSystem, model.realizeDynamics(dummy_state));
+        ASSERT_THROW(ComponentHasNoSystem, 
+                     model.computeStateVariableDerivatives(dummy_state));
+        // should be able to create a Manager either
+        ASSERT_THROW(ComponentHasNoSystem,
+                     Manager manager(model));
+
+        // get a valid state
+        SimTK::State state = model.initSystem();
+        Manager manager(model);
+        state.setTime(0.0);
+
+        // this should invalidate the System
+        model.finalizeFromProperties();
+
+        // verify that finalizeFromProperties() wipes out the underlying System
+        ASSERT_THROW(ComponentHasNoSystem, model.getSystem());
+
+        // should not be able to "trick" the manager into integrating a model
+        // given a stale but compatible state
+        ASSERT_THROW(ComponentHasNoSystem, manager.integrate(state, 1.));
+
+        state = model.initSystem();
+        //re-establish all the connections in the model
+        model.finalizeConnections(model);
```

Comment:
Why is `model.finalizeConnections(model);` required here? Won't `model.initSystem();` on L89 already have "re-establish[ed] all the connections in the model"?